posit-dev/positron
# Notebooks/basic html output support

Addresses #4018 

- Add basic support for HTML output, aka things like pandas dataframe tables. 
- Only show a single output element per cell output. 
  - A cell output can have multiple data bundles of `mime` + `data`, previously we would render all of them. Now we just choose one to render, like other notebook renderers. 
  - Currently, this is done via a fairly arbitrary ordering of "interesting" output types, but in the future we may be more structured about it. 

<img width="1199" alt="image" src="https://github.com/user-attachments/assets/3e9b7e89-855c-48c0-82a3-dcf6ebe4b823">

Also includes a large shuffling of the positron instance code to better organize it according to how other code in positron is organized with private and public fields separated. 


### QA Notes

Notebooks should continue to work as normal, just simple html outputs will now actually render. E.g. the above table. Here's the code to generate one:
```python
import pandas as pd
import numpy as np

# Generate random data
pd.DataFrame({
    'A': np.random.randint(0, 100, size=20),
    'B': np.random.normal(0, 1, size=20),
    'C': np.random.choice(['X', 'Y', 'Z'], size=20),
    'D': np.random.uniform(0, 1, size=20)
})
```
